qcodes/instrument_drivers/tektronix/Keithley_3706A_Matrix_channels.py

```diff
@@ -0,0 +1,709 @@
+        self.write(f"channel.setbackplane('{val}', '{backplane}')")
+
+    def get_backplane(self, val: str) -> str:
+        """
+        Lists the backplane relays that are controlled in switching
+        applications for specified channels.
+
+        Args:
+            val: A string representing the channels being queried.
+        """
+        backplanes = self.get_analog_backplane_specifiers()
+        specifiers = val.split(',')
+        for element in specifiers:
+            if element in backplanes:
+                raise InvalidValue(f'{val} cannot be a analog backplane relay.')
+        if not self._validator(val):
+            raise InvalidValue(f'{val} is not a valid specifier. '
+                               'The specifier should be channels, channel '
+                               'ranges, slots, or "allslots".')
+        return self.ask(f"channel.getbackplane('{val}')")
+
+    def _get_slot_id(self) -> List[str]:
+        """
+        Returns the slot ids of the installed cards.
+        """
+        cards = self.get_switch_cards()
+        slot_id = []
+        for _, item in enumerate(cards):
+            slot_id.append('{slot_no}'.format(**item))
+        return slot_id
+
+    def _get_slot_name(self) -> List[str]:
+        """
+        Returns the names of the slots as "slotX",
+        where "X" is the slot id.
+        """
+        slot_id = self._get_slot_id()
+        slot_names = [f'slot{x}' for x in slot_id]
+        return slot_names
+
+    def _get_number_of_rows(self) -> List[int]:
+        """
+        Returns the total number of rows of the installed cards.
+        """
+        slot_id = self._get_slot_id()
+        total_number_of_rows = [int(float(self.ask(
+            f'slot[{int(i)}].rows.matrix'))) for i in slot_id]
+        return total_number_of_rows
+
+    def _get_number_of_columns(self) -> List[int]:
+        """
+        Returns the total number of columns of the installed cards.
+        """
+        slot_id = self._get_slot_id()
+        total_number_of_columns = [int(float(self.ask(
+            f'slot[{int(i)}].columns.matrix'))) for i in slot_id]
+        return total_number_of_columns
+
+    def _get_rows(self) -> List[List[str]]:
+        """
+        Returns the elements of each row.
+        """
+        total_number_of_rows = self._get_number_of_rows()
+        row_list = []
+        for _, item in enumerate(total_number_of_rows):
+            rows_in_each_slot = [str(i) for i in range(1, item + 1)]
+            row_list.append(rows_in_each_slot)
+        return row_list
+
+    def _get_columns(self) -> List[List[str]]:
+        """
+        Returns the elements of each column.
+        """
+        total_number_of_columns = self._get_number_of_columns()
+        column_list = []
+        for _, item in enumerate(total_number_of_columns):
+            columns_in_each_slot = []
+            for i in range(1, item + 1):
+                if i < 10:
```

Comment:
there's no need for `if i < 10`, the formatting string will do the trick - `f'{i:02d}'`. (there are a occasions of this in other methods as well)

Reply:
Here I prefer the current implementation --the logic flow-- as it is more readable, controllable and understandable than compact format method. Plus, there will be no performance gain.

Reply:
this is not about performance, this is about "not reinventing the wheel" and using existing proven/tested methods.

alright, at least we have a test that covers this code, right?